### PR TITLE
feat: FEWP-249/fix rudderstack issue

### PR DESCRIPTION
**Pull request by shayan-deriv**
## Changes
- Resolved conflict between CleverTap SDK and application's AMD usage in device mode.
  - Temporarily disable `window.define` before loading RudderStack.
  - Restore original `define` function after all device mode destinations are loaded.
  - This fix allows CleverTap to initialize properly without interfering with the application's existing AMD setup.

### CODE REVIEW

**Review by thisyahlen-deriv — APPROVED** [review]

**Review by adrienne-deriv — APPROVED** [review]
LGTM

File: src/rudderstack.ts [hunk 1]
```diff
@@ -74,11 +74,14 @@ export class RudderStack {
 
     init = (RUDDERSTACK_KEY: string) => {
         if (RUDDERSTACK_KEY) {
+            const _define = window.define
```
Comment (ali-hosseini-deriv) [inline src/rudderstack.ts:77]: @shayan-deriv my concern is that we are trying to fix a problem only for Outsystems in a package that is getting used in different Frameworks and not only in Outsystems.
Will this cause any issue?

Comment (shayan-deriv) [reply src/rudderstack.ts:77]: good point. then how about passing a flag for having this logic(which is false by default) ? 🤔

Comment (ali-hosseini-deriv) [reply src/rudderstack.ts:77]: what kind of flag?

Comment (shayan-deriv) [reply src/rudderstack.ts:77]: like this:

```
init = (RUDDERSTACK_KEY: string, disableAMD: boolean = false) => {
        if (RUDDERSTACK_KEY) {
            let _define: any;
            if (disableAMD) {
                _define = window.define;
                window.define = undefined;
            }

            this.setCookieIfNotExists();
            this.analytics.load(RUDDERSTACK_KEY, 'https://deriv-dataplane.rudderstack.com', {
                externalAnonymousIdCookieName: this.rudderstack_anonymous_cookie_key,
            });
            this.analytics.ready(() => {
                if (disableAMD) {
                    window.define = _define;
                }
                this.has_initialized = true;
                this.has_identified = !!(this.getUserId() || this.getAnonymousId());
                this.handleCachedEvents();
            });
        }
    }
    ```

**Review by amam-deriv — APPROVED** [review]

**Review by ali-hosseini-deriv — APPROVED** [review]

**Pull request merged**